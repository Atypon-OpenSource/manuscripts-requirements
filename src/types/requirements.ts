/*!
 * Â© 2020 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { SectionNode } from '@manuscripts/manuscript-transform'
import {
  Model,
  ObjectTypes,
  Section,
  SectionDescription,
} from '@manuscripts/manuscripts-json-schema'

export type TemplateRequirements = Partial<Record<ObjectTypes, Model[]>>

export type CountValidationType =
  | 'manuscript-maximum-characters'
  | 'manuscript-minimum-characters'
  | 'manuscript-maximum-words'
  | 'manuscript-minimum-words'
  | 'section-maximum-characters'
  | 'section-minimum-characters'
  | 'section-maximum-words'
  | 'section-minimum-words'
  | 'manuscript-title-maximum-characters'
  | 'manuscript-title-minimum-characters'
  | 'manuscript-title-maximum-words'
  | 'manuscript-title-minimum-words'
  | 'manuscript-maximum-figures'
  | 'manuscript-maximum-tables'
  | 'manuscript-maximum-combined-figure-tables'
  | 'manuscript-maximum-references'
  | 'figure-minimum-width-resolution'
  | 'figure-minimum-height-resolution'
  | 'figure-maximum-width-resolution'
  | 'figure-maximum-height-resolution'

export type FigureValidationType = 'figure-format-validation'

export interface RequiredSectionValidationResult extends BaseValidationResult {
  type: 'required-section'
  data: { sectionDescription: SectionDescription }
}

export interface SectionTitleValidationResult extends BaseValidationResult {
  type: 'section-title-match' | 'section-title-contains-content'
  data: { id: string; title?: string }
}

export interface SectionOrderValidationResult extends BaseValidationResult {
  type: 'section-order'
  data: { order: Array<string> }
}

export interface SectionBodyValidationResult extends BaseValidationResult {
  type: 'section-body-has-content'
  data: { id: string }
}

export interface SectionCategoryValidation extends BaseValidationResult {
  type: 'section-category-uniqueness'
  data: { id: string }
}
export interface CountValidationResult extends BaseValidationResult {
  type: CountValidationType
  data: { count: number; value: number; id?: string }
  sectionCategory?: string
}

export interface FigureFormatValidationResult extends BaseValidationResult {
  type: FigureValidationType
  data: { id: string; contentType: string }
}

export interface BibliographyValidationResult extends BaseValidationResult {
  type: 'bibliography-doi-format' | 'bibliography-doi-exist'
  data: { id: string }
}

export type BaseValidationResult = {
  type: string
  passed: boolean
  severity: number
}
export type ValidationResult =
  | RequiredSectionValidationResult
  | CountValidationResult
  | FigureFormatValidationResult
  | SectionTitleValidationResult
  | SectionOrderValidationResult
  | SectionBodyValidationResult
  | SectionCategoryValidation
  | BibliographyValidationResult

export type RequiredSections = Array<{
  sectionDescription: SectionDescription
  severity: number
}>

export type CountRequirement = { count: number | undefined; severity: number }

export type CountRequirements = {
  characters: {
    max?: CountRequirement
    min?: CountRequirement
  }
  words: {
    max?: CountRequirement
    min?: CountRequirement
  }
  title?: {
    max?: CountRequirement
    min?: CountRequirement
  }
}

export type FigureCountRequirements = {
  figures: {
    max?: CountRequirement
  }
}

export type CombinedFigureTableCountRequirements = {
  combinedFigureTable: {
    max?: CountRequirement
  }
}

export type TableCountRequirements = {
  tables: {
    max?: CountRequirement
  }
}

export type SectionCountRequirements = Record<string, CountRequirements>

export type SectionTitleRequirement = {
  title: string
  severity: number
  category: string
}

export type SectionWithTitle = {
  title: string
  id: string
}
export interface CountRequirementModel extends Model {
  count?: number
  ignored?: boolean
  severity: number
}

export type SectionDescriptionCountProperty =
  | 'maxWordCount'
  | 'minWordCount'
  | 'maxKeywordCount'
  | 'minKeywordCount'
  | 'maxCharCount'
  | 'minCharCount'

export interface Counts {
  words: number
  characters: number
}

export type ReferenceCountRequirements = {
  references: {
    max?: CountRequirement
  }
}

export type Sections = Map<
  string,
  Array<{ node: SectionNode; counts: Counts; section: Section }>
>

export type FigureResolutionsRequirements = {
  max: {
    width?: CountRequirement
    height?: CountRequirement
  }
  min: {
    width?: CountRequirement
    height?: CountRequirement
  }
}
