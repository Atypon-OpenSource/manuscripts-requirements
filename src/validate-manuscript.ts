/*!
 * Â© 2020 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { ManuscriptTemplate, SectionCategory } from '@manuscripts/json-schema'

import { InputError } from './errors'
import { createRequirementsValidator } from './validate'

const buildSectionCategoriesMap = (
  sectionCategory: SectionCategory[],
  map: Map<string, SectionCategory>
) => {
  for (const category of sectionCategory) {
    map.set(category._id, category)
  }
}

export const sectionCategoriesMap = new Map<string, SectionCategory>()
export let template: ManuscriptTemplate

export const createTemplateValidator = (
  manuscriptTemplate: ManuscriptTemplate | null,
  sectionCategories: SectionCategory[]
) => {
  if (!manuscriptTemplate) {
    throw new InputError('Could not find template')
  }

  buildSectionCategoriesMap(sectionCategories, sectionCategoriesMap)

  template = manuscriptTemplate
  return createRequirementsValidator(manuscriptTemplate as ManuscriptTemplate)
}
