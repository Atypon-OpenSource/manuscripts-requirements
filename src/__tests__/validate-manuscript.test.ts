/*!
 * Â© 2020 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import 'regenerator-runtime/runtime'

import { ContainedModel } from '@manuscripts/transform'
import fs from 'fs'

import { validationOptions } from '../types/input'
import { createTemplateValidator } from '../validate-manuscript'
import { data } from './__fixtures__/manuscript-data.json'

describe('validate manuscript', () => {
  test('validate manuscript', async () => {
    const manuscriptsData = data as unknown as Array<ContainedModel>
    const results = await validate(manuscriptsData)
    results.forEach((result) => {
      // @ts-ignore
      expect(result).toMatchSnapshot(
        {
          _id: expect.any(String),
        },
        'validate-manuscript'
      )
    })
  })

  test('validate manuscript with ignored results', async () => {
    const manuscriptsData = data as unknown as Array<ContainedModel>
    const results = await validate(manuscriptsData)
    manuscriptsData.push(
      // @ts-ignore
      ...results.map((result) => ({ ...result, ignored: true }))
    )
    const newResult = await validate(manuscriptsData)

    expect(newResult.length).toEqual(0)
  })

  test('validate manuscript with options', async () => {
    const manuscriptsData = data as unknown as Array<ContainedModel>
    const results = await validate(manuscriptsData, {
      validateImageFiles: false,
    })
    const figureTypes = [
      'figure-minimum-width-resolution',
      'figure-minimum-height-resolution',
      'figure-maximum-width-resolution',
      'figure-maximum-height-resolution',
      'figure-contains-image',
      'figure-format-validation',
    ]
    results.forEach((result) => {
      // @ts-ignore
      expect(figureTypes.toString()).toEqual(
        expect.not.stringContaining(result.type as string)
      )
    })
  })
})

const validate = async (
  data: Array<ContainedModel>,
  options?: validationOptions
) => {
  const validateManuscript = createTemplateValidator(
    {
      _id: 'MPManuscriptTemplate:www-zotero-org-styles-nature-genetics-Nature-Genetics-Journal-Publication-Article',
      objectType: 'MPManuscriptTemplate',
      bundle: 'MPBundle:www-zotero-org-styles-american-medical-association',
      title: 'Article',
      createdAt: 1433095333.219065,
      updatedAt: 1433095363.405134,
      containerID: 'MPProject:1',
    },
    []
  )

  const getData = async (id: string): Promise<Buffer | undefined> => {
    const path = `${__dirname}/__fixtures__/data/${id}`
    if (fs.existsSync(path)) {
      return fs.readFileSync(path)
    }
    return undefined
  }
  return await validateManuscript(
    data,
    'MPManuscript:9E0BEDBC-1084-4AA1-AB82-10ACFAE02232',
    getData,
    options
  )
}
