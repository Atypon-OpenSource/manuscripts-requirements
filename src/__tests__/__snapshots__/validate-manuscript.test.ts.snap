// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate manuscript validate manuscript: validate-manuscript 1`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "sectionCategory": "MPSectionCategory:introduction",
    "sectionDescription": Object {
      "_id": "MPSectionDescription:30856582-BC9E-426A-B4C5-56F5F681CDF8",
      "objectType": "MPSectionDescription",
      "required": true,
      "sectionCategory": "MPSectionCategory:introduction",
    },
  },
  "fixable": true,
  "ignored": false,
  "message": "There must exist a \\"Introduction\\" section",
  "objectType": "MPRequiredSectionValidationResult",
  "passed": false,
  "severity": 0,
  "type": "required-section",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 2`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "sectionCategory": "MPSectionCategory:materials-method",
    "sectionDescription": Object {
      "_id": "MPSectionDescription:03FC1135-1F04-49A0-9D8F-E595EC2E6863",
      "objectType": "MPSectionDescription",
      "required": true,
      "sectionCategory": "MPSectionCategory:materials-method",
      "title": "online methods",
    },
  },
  "fixable": true,
  "ignored": false,
  "message": "There must exist a \\"Materials & Methods\\" section",
  "objectType": "MPRequiredSectionValidationResult",
  "passed": false,
  "severity": 0,
  "type": "required-section",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 3`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "sectionCategory": "MPSectionCategory:abstract",
    "sectionDescription": Object {
      "_id": "MPSectionDescription:9F8AC4A6-0B93-46A9-ADF4-BC8F6A6603BE",
      "maxParagraphsCount": 1,
      "maxTitleCharLength": 400,
      "maxWordCount": 400,
      "objectType": "MPSectionDescription",
      "required": true,
      "sectionCategory": "MPSectionCategory:abstract",
    },
  },
  "fixable": true,
  "ignored": false,
  "message": "There must exist a \\"Abstract\\" section",
  "objectType": "MPRequiredSectionValidationResult",
  "passed": true,
  "severity": 0,
  "type": "required-section",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 4`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "sectionCategory": "MPSectionCategory:results",
    "sectionDescription": Object {
      "_id": "MPSectionDescription:6C7F0155-A41D-456E-B396-E06FEB081CA9",
      "objectType": "MPSectionDescription",
      "required": true,
      "sectionCategory": "MPSectionCategory:results",
      "title": "Results",
    },
  },
  "fixable": true,
  "ignored": false,
  "message": "There must exist a \\"Results\\" section",
  "objectType": "MPRequiredSectionValidationResult",
  "passed": false,
  "severity": 0,
  "type": "required-section",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 5`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "sectionCategory": "MPSectionCategory:competing-interests",
    "sectionDescription": Object {
      "_id": "MPSectionDescription:C37FB688-B8E2-44A3-959A-C76CA1C405A3",
      "objectType": "MPSectionDescription",
      "required": true,
      "sectionCategory": "MPSectionCategory:competing-interests",
      "title": "Competing financial interest",
    },
  },
  "fixable": true,
  "ignored": false,
  "message": "There must exist a \\"Competing Interests\\" section",
  "objectType": "MPRequiredSectionValidationResult",
  "passed": true,
  "severity": 0,
  "type": "required-section",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 6`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "sectionCategory": "MPSectionCategory:discussion",
    "sectionDescription": Object {
      "_id": "MPSectionDescription:43239655-328D-4A07-A587-B21BF8E4230F",
      "objectType": "MPSectionDescription",
      "required": true,
      "sectionCategory": "MPSectionCategory:discussion",
      "title": "Discussion",
    },
  },
  "fixable": true,
  "ignored": false,
  "message": "There must exist a \\"Discussion\\" section",
  "objectType": "MPRequiredSectionValidationResult",
  "passed": false,
  "severity": 0,
  "type": "required-section",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 7`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "sectionCategory": "MPSectionCategory:keywords",
    "sectionDescription": Object {
      "_id": "MPSectionDescription:B16D5504-DB50-4FE4-B2EE-6B77223CE511",
      "maxKeywordCount": 6,
      "minKeywordCount": 4,
      "objectType": "MPSectionDescription",
      "required": true,
      "sectionCategory": "MPSectionCategory:keywords",
    },
  },
  "fixable": true,
  "ignored": false,
  "message": "There must exist a \\"Keywords\\" section",
  "objectType": "MPRequiredSectionValidationResult",
  "passed": false,
  "severity": 0,
  "type": "required-section",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 8`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 510,
    "value": 400,
  },
  "ignored": false,
  "message": "The manuscript must have less than or equal to 400 characters",
  "objectType": "MPCountValidationResult",
  "passed": false,
  "severity": 0,
  "type": "manuscript-maximum-characters",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 9`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 510,
    "value": 100,
  },
  "ignored": false,
  "message": "The manuscript has more than or equal to 100 characters",
  "objectType": "MPCountValidationResult",
  "passed": true,
  "severity": 0,
  "type": "manuscript-minimum-characters",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 10`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 90,
    "value": 4000,
  },
  "ignored": false,
  "message": "The manuscript has less than or equal to 4000 words",
  "objectType": "MPCountValidationResult",
  "passed": true,
  "severity": 0,
  "type": "manuscript-maximum-words",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 11`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 90,
    "value": 2000,
  },
  "ignored": false,
  "message": "The manuscript must have more than or equal to 2000 words",
  "objectType": "MPCountValidationResult",
  "passed": false,
  "severity": 0,
  "type": "manuscript-minimum-words",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 12`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 14,
    "sectionCategory": "MPSectionCategory:abstract",
    "value": 400,
  },
  "ignored": false,
  "message": "\\"Abstract\\" has less than or equal to 400 words",
  "objectType": "MPCountValidationResult",
  "passed": true,
  "severity": 0,
  "type": "section-maximum-words",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 13`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 1,
    "sectionCategory": "MPSectionCategory:abstract",
    "value": 1,
  },
  "ignored": false,
  "message": "\\"Abstract\\" has less than or equal to 1 paragraphs",
  "objectType": "MPCountValidationResult",
  "passed": true,
  "severity": 0,
  "type": "section-maximum-paragraphs",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 14`] = `
Object {
  "_id": Any<String>,
  "affectedElementId": "MPSection:7EB59268-1C3A-4522-B250-A7B09AC5C977",
  "data": Object {
    "sectionCategory": "MPSectionCategory:competing-interests",
  },
  "ignored": false,
  "message": "\\"Competing Interests\\" title has content (is not empty)",
  "objectType": "MPSectionTitleValidationResult",
  "passed": true,
  "severity": 0,
  "type": "section-title-contains-content",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 15`] = `
Object {
  "_id": Any<String>,
  "affectedElementId": "MPSection:7EB59268-1C3A-4522-B250-A7B09AC5C977",
  "data": Object {
    "sectionCategory": "MPSectionCategory:competing-interests",
    "title": "Competing financial interest",
  },
  "fixable": true,
  "ignored": false,
  "message": "Title for \\"Competing Interests\\" section should be \\"Competing financial interest\\"",
  "objectType": "MPSectionTitleValidationResult",
  "passed": false,
  "severity": 0,
  "type": "section-title-match",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 16`] = `
Object {
  "_id": Any<String>,
  "affectedElementId": "MPSection:A67FABCE-B7A0-457A-95C0-3A6B4AFD80BB",
  "data": Object {
    "sectionCategory": "MPSectionCategory:abstract",
    "sectionTitle": "A paragraph",
  },
  "ignored": false,
  "message": "\\"Abstract\\" section has content (is not empty)",
  "objectType": "MPSectionBodyValidationResult",
  "passed": true,
  "severity": 0,
  "type": "section-body-has-content",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 17`] = `
Object {
  "_id": Any<String>,
  "affectedElementId": "MPSection:7EB59268-1C3A-4522-B250-A7B09AC5C977",
  "data": Object {
    "sectionCategory": "MPSectionCategory:competing-interests",
    "sectionTitle": "test",
  },
  "ignored": false,
  "message": "\\"Competing Interests\\" section has content (is not empty)",
  "objectType": "MPSectionBodyValidationResult",
  "passed": true,
  "severity": 0,
  "type": "section-body-has-content",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 18`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 3,
    "value": 50,
  },
  "ignored": false,
  "message": "The manuscript title has less than or equal to 50 words",
  "objectType": "MPCountValidationResult",
  "passed": true,
  "severity": 0,
  "type": "manuscript-title-maximum-words",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 19`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 3,
    "value": 10,
  },
  "ignored": false,
  "message": "The manuscript title must have more than or equal to 10 words",
  "objectType": "MPCountValidationResult",
  "passed": false,
  "severity": 0,
  "type": "manuscript-title-minimum-words",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 20`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 16,
    "value": 80,
  },
  "ignored": false,
  "message": "The manuscript title has less than or equal to 80 characters",
  "objectType": "MPCountValidationResult",
  "passed": true,
  "severity": 0,
  "type": "manuscript-title-maximum-characters",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 21`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 16,
    "value": 20,
  },
  "ignored": false,
  "message": "The manuscript title must have more than or equal to 20 characters",
  "objectType": "MPCountValidationResult",
  "passed": false,
  "severity": 0,
  "type": "manuscript-title-minimum-characters",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 22`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 1,
    "value": 15,
  },
  "ignored": false,
  "message": "The manuscript has less than or equal to 15 references",
  "objectType": "MPCountValidationResult",
  "passed": true,
  "severity": 0,
  "type": "manuscript-maximum-references",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 23`] = `
Object {
  "_id": Any<String>,
  "affectedElementId": "MPBibliographyItem:6C6AF046-1C45-4CC5-9F38-93BC736E5293",
  "ignored": false,
  "message": "DOI is required for bibliographic references",
  "objectType": "MPBibliographyValidationResult",
  "passed": false,
  "severity": 0,
  "type": "bibliography-doi-exist",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 24`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 2,
    "value": 5,
  },
  "ignored": false,
  "message": "The manuscript has less than or equal to 5 figures",
  "objectType": "MPCountValidationResult",
  "passed": true,
  "severity": 0,
  "type": "manuscript-maximum-figures",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 25`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 1,
    "value": 3,
  },
  "ignored": false,
  "message": "The manuscript has less than or equal to 3 tables",
  "objectType": "MPCountValidationResult",
  "passed": true,
  "severity": 0,
  "type": "manuscript-maximum-tables",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 26`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 3,
    "value": 3,
  },
  "ignored": false,
  "message": "Requirement passed",
  "objectType": "MPCountValidationResult",
  "passed": true,
  "severity": 0,
  "type": "manuscript-maximum-combined-figure-tables",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 27`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "allowedImageTypes": Array [
      "tiff",
      "jpeg",
    ],
    "contentType": "image/jpeg",
  },
  "ignored": false,
  "message": "Required image file format (JPEG)",
  "objectType": "MPFigureFormatValidationResult",
  "passed": true,
  "severity": 0,
  "type": "figure-format-validation",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 28`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "allowedImageTypes": Array [
      "tiff",
      "jpeg",
    ],
    "contentType": "image/png",
  },
  "ignored": false,
  "message": "PNG format is not allowed, allowed formats (TIFF,JPEG)",
  "objectType": "MPFigureFormatValidationResult",
  "passed": false,
  "severity": 0,
  "type": "figure-format-validation",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 29`] = `
Object {
  "_id": Any<String>,
  "affectedElementId": "MPFigure:C9A3CBCE-4AF4-4AD2-97B0-D2848BEFDE5F",
  "ignored": false,
  "message": "Image data for figure is missing",
  "objectType": "MPFigureImageValidationResult",
  "passed": false,
  "severity": 0,
  "type": "figure-contains-image",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 30`] = `
Object {
  "_id": Any<String>,
  "affectedElementId": "MPFigure:C9A3CBCE-4AF4-4AD2-97B0-D2848BEFDE5A",
  "ignored": false,
  "message": "Image data for figure is included",
  "objectType": "MPFigureImageValidationResult",
  "passed": true,
  "severity": 0,
  "type": "figure-contains-image",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 31`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 304,
    "dpi": undefined,
    "id": "MPFigure:C9A3CBCE-4AF4-4AD2-97B0-D2848BEFDE5A",
    "value": 600,
  },
  "ignored": false,
  "message": "Figure width must be greater than or equal to (600px)",
  "objectType": "MPFigureResolution",
  "passed": false,
  "severity": 0,
  "type": "figure-minimum-width-resolution",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 32`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 304,
    "dpi": undefined,
    "id": "MPFigure:C9A3CBCE-4AF4-4AD2-97B0-D2848BEFDE5A",
    "value": 4000,
  },
  "ignored": false,
  "message": "Figure width is less than or equal to (4000px)",
  "objectType": "MPFigureResolution",
  "passed": true,
  "severity": 0,
  "type": "figure-maximum-width-resolution",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 33`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 350,
    "dpi": undefined,
    "id": "MPFigure:C9A3CBCE-4AF4-4AD2-97B0-D2848BEFDE5A",
    "value": 400,
  },
  "ignored": false,
  "message": "Figure height must be greater than or equal to (400px)",
  "objectType": "MPFigureResolution",
  "passed": false,
  "severity": 0,
  "type": "figure-minimum-height-resolution",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 34`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 350,
    "dpi": undefined,
    "id": "MPFigure:C9A3CBCE-4AF4-4AD2-97B0-D2848BEFDE5A",
    "value": 1000,
  },
  "ignored": false,
  "message": "Figure height is less than or equal to (1000px)",
  "objectType": "MPFigureResolution",
  "passed": true,
  "severity": 0,
  "type": "figure-maximum-height-resolution",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 35`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 2,
    "value": 1,
  },
  "ignored": false,
  "message": "The manuscript must have less than or equal to 1 corresponding authors",
  "objectType": "MPCountValidationResult",
  "passed": false,
  "severity": 0,
  "type": "manuscript-maximum-corresponding-authors",
}
`;

exports[`validate manuscript validate manuscript: validate-manuscript 36`] = `
Object {
  "_id": Any<String>,
  "data": Object {
    "count": 111,
    "value": 60,
  },
  "ignored": false,
  "message": "The manuscript running title must have less than or equal to 60 characters",
  "objectType": "MPCountValidationResult",
  "passed": false,
  "severity": 0,
  "type": "manuscript-running-title-maximum-characters",
}
`;
